Mu2eKinKal/src/LoopHelixFit_module.cc

```diff
@@ -165,11 +168,16 @@ namespace mu2e {
       explicit LoopHelixFit(const Parameters& settings);
       void beginRun(art::Run& run) override;
       void produce(art::Event& event) override;
+      void endJob() override;
     private:
       TrkFitFlag fitflag_;
       // parameter-specific functions that need to be overridden in subclasses
       KTRAJ makeSeedTraj(HelixSeed const& hseed,TimeRange const& trange,VEC3 const& bnom, int charge) const;
       bool goodFit(KKTRK const& ktrk) const;
+      bool goodHelix(HelixSeed const& hseed) const;
+      std::vector<HelixRecoDir::PropDir> chooseHelixDir(HelixSeed const& hseed) const;
+      int compare_tracks(const std::unique_ptr<KKTRK>& ktrk_1, const std::unique_ptr<KKTRK>& ktrk_2) const;
+      std::unique_ptr<KKTRK> fitTrack(art::Event& event, HelixSeed const& hseed, const float dir, const PDGCode::type fitpart);
```

Comment:
I'm unclear why you pass dir as a float. Using TrkFitDirection would make the interface unambiguous. If you need the FP direction (line 413) you can get that from TrkFitDirection::dzdt()

Reply:
I updated to passing the HelixRecoDir::PropDir enum directly, and then checking this enum to decide on the slope direction.